backend/app/models/anti_recommendations_selector.py

```diff
@@ -0,0 +1,19 @@
+from typing import NamedTuple
+
+
+class AntiRecommendationsSelector:
```

Comment:
Nit: probably would have done a match expression on an enum as the selector, since it'd be more opaque to callers.

Reply:
`AntiRecommendationsSelector` has been changed to an enum, and `UserService` uses a match expression to remove anti-recommendations.